packages/block-editor/src/components/link-control/index.js

```diff
@@ -203,14 +254,42 @@ function LinkControl( {
 		// If it's potentially a URL search then concat on a URL search suggestion
 		// just for good measure. That way once the actual results run out we always
 		// have a URL option to fallback on.
-		return couldBeURL && ! args.isInitialSuggestions
-			? results[ 0 ].concat( results[ 1 ] )
-			: results[ 0 ];
+		results =
+			couldBeURL && ! args.isInitialSuggestions
+				? results[ 0 ].concat( results[ 1 ] )
+				: results[ 0 ];
+
+		// Here we append a faux suggestion to represent a "CREATE" option. This
+		// is detected in the rendering of the search results and handled as a
+		// special case. This is currently necessary because the suggestions
+		// dropdown will only appear if there are valid suggestions and
+		// therefore unless the create option is a suggestion it will not
+		// display in scenarios where there are no results returned from the
+		// API. In addition promoting CREATE to a first class suggestion affords
+		// the a11y benefits afforded by `URLInput` to all suggestions (eg:
+		// keyboard handling, ARIA roles...etc).
+		//
+		// Note also that the value of the `title` and `url` properties must correspond
+		// to the text value of the `<input>`. This is because `title` is used
+		// when creating the entity. Similarly `url` is used when using keyboard to select
+		// the suggestion (the <form> `onSubmit` handler falls-back to `url`).
+		return maybeURL( val )
+			? results
+			: results.concat( {
+					// the `id` prop is intentionally ommitted here because it
+					// is never exposed as part of the component's public API.
+					// see: https://github.com/WordPress/gutenberg/pull/19775#discussion_r378931316.
+					title: val, // must match the existing `<input>`s text value
+					url: val, // must match the existing `<input>`s text value
+					type: CREATE_TYPE,
+			  } );
 	};
 
 	/**
+	 * Determines which type of search handler to use based on the users input.
 	 * Cancels editing state and marks that focus may need to be restored after
 	 * the next render, if focus was within the wrapper when editing finished.
+	 * else a API search should made for matching Entities.
```

Comment:
The `else` sentence may have gotten cut off from a rebase or something.

Reply:
Good spot. Totally broken comment. Thanks rebase!